### PR TITLE
Add integration tests for VersionedLayerClient

### PR DESCRIPTION
**Pull request by diachenko-mischa**
Copy integration tests from CatalogClient to VersionedLayerClient that
are relevant and use deprecated API.

Relates-To: OLPEDGE-984
Signed-off-by: Diachenko Mykahilo <ext-mykhailo.z.diachenko@here.com>

### CODE REVIEW

File: tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp [hunk 1]
```diff
@@ -1866,4 +1902,691 @@ TEST_F(DataserviceReadVersionedLayerClientTest,
...
+  EXPECT_CALL(
+      *network_mock_,
+      Send(IsGetRequest("https://blob-ireland.data.api.platform.here.com/"
+                        "blobstore/v1/catalogs/hereos-internal-test-v2/"
+                        "layers/testlayer/data/invalidDataHandle"),
+           _, _, _, _))
+      .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(404),
+                                   "Resource not found."));
+
+  auto catalog_client =
```
Comment (borzun) [inline tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp]:
```suggestion
  auto client =
```
Also in other places

File: tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp [hunk 2]
```diff
@@ -247,6 +248,41 @@ class DataserviceReadVersionedLayerClientTest : public ::testing::Test {
                                    olp::http::HttpStatusCode::OK),
                                HTTP_RESPONSE_BLOB_DATA_PREFETCH_6));
 
+    ON_CALL(*network_mock_, Send(IsGetRequest(URL_BLOB_DATA_269), _, _, _, _))
+        .WillByDefault(
+            ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(200),
```
Comment (borzun) [inline tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp]:
```suggestion
            ReturnHttpResponse(olp::http::NetworkResponse().WithStatus( olp::http::HttpStatusCode::OK),
```
Also in other places.

File: tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp [hunk 3]
```diff
@@ -247,6 +248,41 @@ class DataserviceReadVersionedLayerClientTest : public ::testing::Test {
...
+    ON_CALL(*network_mock_,
+            Send(IsGetRequest(URL_BLOB_DATA_269_V2), _, _, _, _))
+        .WillByDefault(
+            ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(200),
+                               HTTP_RESPONSE_BLOB_DATA_269_V2));
+
+    ON_CALL(*network_mock_,
+            Send(IsGetRequest(URL_QUERY_PARTITION_269_V10), _, _, _, _))
+        .WillByDefault(
+            ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(400),
```
Comment (borzun) [inline tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp]:
```suggestion
            ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(olp::http::HttpStatusCode::BAD_REQUEST),
```
Also in other places.

File: tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp [hunk 4]
```diff
@@ -24,6 +24,7 @@
 #include <matchers/NetworkUrlMatchers.h>
 #include <mocks/NetworkMock.h>
 #include <olp/authentication/Settings.h>
+#include <olp/core/cache/DefaultCache.h>
```
Comment (borzun) [inline tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp]: Are you sure this include required?

I don't see usage of `DefaultCache` or other members of the _DefaultCache.h_ file.

Comment (diachenko-mischa) [reply tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp]: Yes, you're right! This is leftovers.

**Pull request merged**